docs/user-guide.md

```diff
@@ -0,0 +1,856 @@
+       interpreter's environment (accessed via the `pyscript` module).
+    6. Finally, use the interpreter in the correctly configured environment to
+       evaluate the detected Python code.
+* When an interpreter is ready the `py:ready` or `mpy:ready` events are
+  dispatched, depending which interpreter you've specified (Pyodide or
+  MicroPython respectively).
+
+!!! warning
+
+    When using workers, each worker will have a completely separate environment
+    to that of the main thread.
+
+    As a result, configuration in the main thread will be different to that for
+    an interpreter running on a worker. In fact, you can use different
+    interpreters and configuration in each context (for instance, MicroPython
+    on the main thread, and Pyodide on a worker).
+
+Finally, various "hooks" are called at different moments in the lifecycle of
+PyScript. These can be used by plugin authors to modify or enhance the
+behaviour of PyScript. The hooks, and how to use them, are explored further in
+[the section on plugins](#plugins).
+
+## Interpreters
+
+Python is an interpreted language, and thus needs an interpreter to work.
+
+PyScript supports two versions of the Python interpreter that have
+been compiled to WASM: Pyodide and MicroPython. You should select which one to
+use depending on your use case and acceptable trade-offs.
+
+Both interpreters make use of [emscripten](https://emscripten.org/), a compiler
+toolchain (using [LLVM](https://llvm.org/)), for emitting WASM assets for the
+browser. Emscripten also provides APIs so operating-system level features such
+as a sandboxed [file system](https://emscripten.org/docs/api_reference/Filesystem-API.html)
+(**not** the user's local machine's filesystem), [IO](https://emscripten.org/docs/api_reference/console.h.html`)
+(`stdin`, `stdout`, `stderr` etc,) and
+[networking](https://emscripten.org/docs/api_reference/fetch.html) are
+available within the context of a browser.
+
+Both Pyodide and MicroPython implement the same robust
+[Python](https://pyodide.org/en/stable/usage/api/python-api.html)
+⟺ [JavaScript](https://pyodide.org/en/stable/usage/api/js-api.html)
+[foreign function interface](#ffi) (FFI). This
+bridges the gap between the browser and Python worlds.
+
+### Pyodide
+
+<a href="https://pyodide.org/"><img src="../assets/images/pyodide.png"/></a>
+
+[Pyodide](https://pyodide.org/) is a version of the standard
+[CPython](https://python.org/) interpreter, patched to compile to WASM and
+work in the browser.
+
+It includes many useful features:
+
+* The installation of pure Python packages from [PyPI](https://pypi.org/) via
+  the [micropip](https://micropip.pyodide.org/en/stable/index.html) package
+  installer. Some packages with C extensions have versions compiled for WASM
+  and these can also be installed with `micropip`. There are plans afoot to
+  make WASM a target in PyPI so packages with C extenions can be automatically
+  compiled to WASM.
```

Comment:
if you don't understand what are pure python packages and C extension, it is probably not very easy to understand what this paragraph means in practice.
It's probably worth its own section

Reply:
Sure, we do point out that we expect not-beginner-level readers of this guide. Suggestions for a good article to which we can point folks who want to know more are most welcome. :+1:

Reply:
from my experience of talking about C extensions, PyPy, HPy, etc. at conferences, there is a huge number of "non beginners" who don't know anything about C extensions, and that have no idea of what is a "pure python" wheel.

A quick fix is to link to this pyodide's FAQ:
https://pyodide.org/en/stable/usage/faq.html#why-can-t-micropip-find-a-pure-python-wheel-for-a-package

A more proper fix is to write a dedicated section to explain it